src/main/java/com/minecolonies/coremod/entity/ai/util/AITarget.java

```diff
@@ -24,78 +24,182 @@
      */
     private boolean okayToEat;
 
     /**
      * Construct a target.
-     *
+     * TODO: Remove once all Targets transitioned to tickRate
      * @param action the action to apply
      */
     public AITarget(@NotNull final Supplier<AIState> action, final boolean isOkayToEat)
     {
-        this(() -> true, isOkayToEat, action);
+        this(() -> true, isOkayToEat, action, 1);
     }
 
     /**
      * Construct a target.
      *
+     * @param action the action to apply
+     */
+    public AITarget(@NotNull final Supplier<AIState> action, final boolean isOkayToEat, @NotNull final int tickRate)
+    {
+        this(() -> true, isOkayToEat, action, tickRate);
+    }
+
+    /**
+     * Construct a target.
+     * TODO: Remove once all Targets transitioned to tickRate
      * @param predicate the predicate for execution
      * @param action    the action to apply
      */
     public AITarget(@NotNull final BooleanSupplier predicate, final boolean isOkayToEat, @NotNull final Supplier<AIState> action)
     {
-        this(null, isOkayToEat, predicate, action);
+        this(null, isOkayToEat, predicate, action, 1);
     }
 
     /**
      * Construct a target.
      *
+     * @param predicate the predicate for execution
+     * @param action    the action to apply
+     */
+    public AITarget(@NotNull final BooleanSupplier predicate, final boolean isOkayToEat, @NotNull final Supplier<AIState> action, @NotNull final int tickRate)
+    {
+        this(null, isOkayToEat, predicate, action, tickRate);
+    }
+
+    /**
+     * Construct a target.
+     * TODO: Remove once all Targets transitioned to tickRate
      * @param state     the state it needs to be | null
      * @param predicate the predicate for execution
      * @param action    the action to apply
      */
-    public AITarget(@Nullable final AIState state, final boolean isOkayToEat, @NotNull final BooleanSupplier predicate, @NotNull final Supplier<AIState> action)
+    public AITarget(
+      @Nullable final AIState state,
+      final boolean isOkayToEat,
+      @NotNull final BooleanSupplier predicate,
+      @NotNull final Supplier<AIState> action,
+      @NotNull final int tickRate)
     {
-        this.state = state;
+        if (state == null)
+        {
+            this.state = AIState.STATE_BLOCKING_PRIO;
+        }
+        else
+        {
+            this.state = state;
+        }
         this.predicate = predicate;
         this.action = action;
         this.okayToEat = isOkayToEat;
+
+        // Limit rates
+        this.tickRate = tickRate > MAX_AI_TICKRATE ? MAX_AI_TICKRATE : tickRate;
+        this.tickRate = this.tickRate < 1 ? 1 : this.tickRate;
+
+        // Calculate offSet % tickRate already to not have redundant calculations later
+        this.tickOffset = tickOffsetVariant % this.tickRate;
+        // Increase variant for next AITarget and reset variant at a certain point
+        tickOffsetVariant++;
```

Comment:
So this is a global state that offsets actions statically in the order they are created? Is there a good reason for this design choice?

Reply:
the Offsets intent is to not have all AITargets with the same tickrate execute during the same tick and instead shift them up at an equal distribution. E.g. first AITarget with tickrate 10 has an offset of 0 and will be executed on ticks 0,10,20,30 and the second AITarget with tickrate 10 has an offset of 1 and will be executed on ticks 1,11,21,31